stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/EncryptedBackgroundSyncConfigStore.kt

```diff
@@ -4,27 +4,38 @@ import android.content.Context
 import android.content.SharedPreferences
 import androidx.security.crypto.EncryptedSharedPreferences
 import androidx.security.crypto.MasterKey
+import io.getstream.chat.android.client.logger.ChatLogger
 
 private const val MASTER_KEY_ALIAS = "_stream_sync_config_master_key_"
-private const val SYNC_CONFIG_PREFS_NAME = "stream_livedata_sync_config_store"
+private const val ENCRYPTED_SYNC_CONFIG_PREFS_NAME = "stream_livedata_sync_config_store"
+private const val SYNC_CONFIG_PREFS_NAME = ".stream_livedata_sync_config_store"
 private const val KEY_API_KEY = "api_key"
 private const val KEY_USER_ID = "user_id"
 private const val KEY_USER_TOKEN = "user_token"
 
 internal class EncryptedBackgroundSyncConfigStore(context: Context) {
     private val prefs: SharedPreferences
+    private val logger = ChatLogger.get("EncryptedBackgroundSyncConfigStore")
 
     init {
         val masterKey = MasterKey.Builder(context, MASTER_KEY_ALIAS)
             .setKeyScheme(MasterKey.KeyScheme.AES256_GCM)
             .build()
-        prefs = EncryptedSharedPreferences.create(
-            context,
-            SYNC_CONFIG_PREFS_NAME,
-            masterKey,
-            EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
-            EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
-        )
+        prefs = try {
+            EncryptedSharedPreferences.create(
+                context,
+                ENCRYPTED_SYNC_CONFIG_PREFS_NAME,
+                masterKey,
+                EncryptedSharedPreferences.PrefKeyEncryptionScheme.AES256_SIV,
+                EncryptedSharedPreferences.PrefValueEncryptionScheme.AES256_GCM
+            )
+        } catch (e: Exception) {
+            logger.logE("Error creating encrypted shared preferences", e)
+            context.applicationContext.getSharedPreferences(
+                SYNC_CONFIG_PREFS_NAME,
+                Context.MODE_PRIVATE
+            )
```

Comment:
Is this safe enough for our needs? We must be using encrypted SharedPref normally for a reason 😄

Reply:
It stores the userId and the token.
It is into their own app sandbox that is not accessible by other apps (At least they have root permissions)